### ISSUE TITLE
Other class is not completely ignored in IoU calculations

### ISSUE THREAD

**Issue by DavidFM43** [IGNF/FLAIR-2#1]
In the datapaper is mentioned: 

>since the other class is not well-defined and is equivalent to void, it is excluded from the IoU calculations. Consequently, the mIoU is computed as the average of the IoUs from the remaining 12 classes.

However it seems that the predictions on pixels labeled with the `other` class index are not ignored during the calculation of the confusion matrix, which means that these predictions are treated as false positives and affect the value of IoU, Is this the case?. For instance, in [this](
https://github.com/open-mmlab/mmsegmentation/blob/00790766aff22bd6470dbbd9e89ea40685008395/mmseg/evaluation/metrics/iou_metric.py#L185-L187) code for the IoU metric from the popular library [mmsegmentation](https://github.com/open-mmlab/mmsegmentation) those pixels are first ignored before doing the IoU calculations.

**Comment by agarioud** [comment]
Dear @DavidFM43 , 
thank you for your remark and concern. We acknowledge that this sentence in the datapaper is misleading and requires correction. Specifically, we calculate the per-patch confusion matrix and per-class Intersection over Union (IoU) without excluding pixels belonging to the 'other' class, even though they represent a marginal part of the test-set. However, when computing the mean IoU (mIoU), we do remove the IoU of the 'other' class due to its association with majority or lower quality level pixels or very underrepresented land cover.
We chose this approach to avoid potential bias towards methods that might predict more 'other' class pixels when uncertain (e.g., below a threshold on the logits) because they know it won't be considered in the metric calculation.
